crypto/types/types.go

```diff
@@ -5,26 +5,34 @@ import (
 	tmcrypto "github.com/tendermint/tendermint/crypto"
 )
 
-// PubKey interface extends proto.Message
-// and tendermint crypto.PubKey
+// PubKey defines a public key and
+// extends proto.Message.
 type PubKey interface {
 	proto.Message
-	tmcrypto.PubKey
+
+	Address() Address
+	Bytes() []byte
+	VerifySignature(msg []byte, sig []byte) bool
+	Equals(PubKey) bool
+	Type() string
+}
+
+// BasePrivKey defines a private key
+type BasePrivKey interface {
```

Comment:
`BasePrivKey` sounds strange. How about doing other way around and renaming: 
* PrivKey -> ProtoPrivKey
* BasePrivKey -> PrivKey

Reply:
I renamed BasePrivKey to LedgerPrivKey. See comments in 0048063 if they make sense to you.

Reply:
NOTE: this will be removed or changed when `PrivKeyLedgerSecp256k1` will get converted to `proto`